src/__tests__/cli.ts

```diff
@@ -0,0 +1,501 @@
+import fs from 'fs';
+import path from 'path';
+import util from 'util';
+import { CliArguments } from '..';
+
+const mkdir = util.promisify(fs.mkdir);
+const rmdir = util.promisify(fs.rmdir);
+const writeFile = util.promisify(fs.writeFile);
+const readFile = util.promisify(fs.readFile);
+
+async function exec(
+  testProjectDir: string,
+  { init = false, flow = false, update = false }: Partial<CliArguments> = {},
+): Promise<{ exitCode: number | null; stdout: string; stderr: string }> {
+  const originalExit = process.exit;
+  const originalCwd = process.cwd();
+  const originalConsole = {
+    log: console.log,
+    warn: console.warn,
+    error: console.error,
+  };
+
+  try {
+    let exitCode: number | null = null;
+    let stdout = '';
+    let stderr = '';
+
+    const appendStdout = (...args: any[]): void => {
+      stdout += args.map((arg) => arg.toString()).join(' ');
+    };
+
+    const appendStderr = (...args: any[]): void => {
+      stderr += args.map((arg) => arg.toString()).join(' ');
+    };
+
+    console.log = appendStdout;
+    console.warn = appendStdout;
+    console.error = appendStderr;
+
+    process.exit = (code: number): never => {
+      exitCode = exitCode ?? code;
+      return undefined as never;
+    };
+
+    process.chdir(testProjectDir);
+
+    const main = require('..').main as (args: CliArguments) => Promise<void>;
+
+    await main({ init, flow, update });
+
+    return { exitCode: exitCode ?? 0, stdout, stderr };
+  } finally {
+    process.chdir(originalCwd);
+    process.exit = originalExit;
+    Object.entries(originalConsole).forEach(([key, value]) => {
+      console[key] = value;
+    });
+  }
+}
```

Comment:
I'll reiterate my concern here that while this allows the coverage reporting to be more accurate, there is a lot of misdirection and possible mishandling of the test environment here.

My biggest concern is that **`process.exit` does not halt execution** any more, so while the exit code is recorded for testing, the code continues to execute.  In _most_ cases this is fine as the exit call is immediately before the function would naturally return anyway, but there are some cases where this is not the case (which is why `exitCode = exitCode ?? code;` was added to not override the original exit code).